resolwe_bio/processes/icount/segment.yml

```diff
@@ -0,0 +1,62 @@
+- slug: icount-segment
+  name: iCount segment
+  description: Parse genome annotation, into internal iCount format.
+  category: analyses
+  version: 1.0.0
+  type: data:annotation:gtf:icount
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: "resolwe/bio-linux8-resolwe-preview:icount"
+  data_name: "iCount-segment ({{annotation.gtf.file|basename|default('?')}})"
+  flow_collection: sample
+  persistence: CACHED
+  input:
+  - name: annotation
+    label: GTF file with annotation.
+    type: data:annotation:gtf
+  - name: genome
+    label: Data object containing genome (only fai file is used).
+    type: data:seq:nucleotide
+  - name: report_progress
+    label: Switch to print progress to stdout.
+    type: basic:boolean
+    default: false
+  output:
+  - name: gtf
+    label: Segmented file, of custom iCount format.
+    type: basic:file
+  - name: types_length
+    label: File with lengths of regions and subregions (used for iCount annotate)
+    type: basic:file
+  - name: fai
+    label: File with lengths of chromosomes.
+    type: basic:file
+  - name: source
+    label: Gene ID source
+    type: basic:string
+  run:
+    language: bash
+    runtime: polyglot
+    program: |
+      iCount segment {{annotation.gtf.file}} segmentation.gtf {{genome.fai.file}} \
+        {% if report_progress %} --report_progress {% endif %}
+      re-checkrc
+      gzip segmentation.gtf
+      re-save-file gtf segmentation.gtf.gz
+
+
+      cut -f1,2 {{genome.fai.file}} > chrom_lenghts.txt
+      re-save-file fai chrom_lenghts.txt
+      # Make also types_length_file:
+      python3 -c 'from iCount.analysis.summary import make_types_length_file; \
```

Comment:
Probably not. This was a workaround until easier functionality is supported in iCount package. I wanted to avoid making a separate script for this...